### DISCUSSION TITLE
Renaming GitHub Releases assets doesn't allow the existing filename extension to be removed

### DISCUSSION THREAD

**Discussion by zeke** [community/community#84094] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

I recorded a one-minute demo to show what's happening: https://www.loom.com/share/c2afb266eebf41b0b10be552d29aa1b9

Here's a transcript of the video:

> Hello, I think I just found a bug on github.com on the releases page where if you try to rename an existing asset with a file extension, you can't get rid of the file extension. So, what I'm trying to do is go in and remove this default prefix from this file, and change the capitalization, and when I do that, I end up with this strangely pre-fixed and extension filing. The only way that I've been able to work around this is to download these assets to my local machine, rename them to remove the extension, go back and edit the release, drag and drop them onto here, And then to eat these provisional filenames. Hopefully that's a helpful bug report. Thanks for watching. Have a nice day. Have a new year. Bye bye.

**Comment by Nqvinh7603** [comment]
I'm unable to access external content like the video you linked to in the GitHub question. However, based on the description provided, it seems like the issue revolves around renaming existing assets in GitHub Releases without being able to remove the file extension.

The user is experiencing a problem where attempting to remove the file extension while renaming an existing asset on the Releases page results in the extension being retained in the filename. The workaround involves downloading the assets, removing the extension locally, and then re-uploading them without the extension to achieve the desired naming convention.

**Comment by nerdCopter** [comment]
i found this post searching for alternative renaming method rather than API.  Thank you, as i did not realize it was so easy.  That said i tested renaming a file and was able to reproduced what you have experienced.  if i rename the extension, it will add my new name and also append the original extension.
my example is `SITL_b37ce3e.hex` renamed to `SITL_b37ce3e.hexxx` will instead result in `SITL_b37ce3e.hexxx.hex`.

this does indeed seem to be a bug, IMHO.  I wonder if the API suffers this as well, i have not yet tested.

edit: maybe it is by design for security reasons.  like the age-old .exe renamed to .png to cause chaos. :man_shrugging: 
edit 2: i found updating filenames via UI also sent out additional emails about old releases :(
edit 3: Here is the API method. It works; however, i was told it would retain download statistics but in-fact, it did not. https://docs.github.com/en/rest/releases/assets?apiVersion=2022-11-28#update-a-release-asset (note: there are two json `id`'s in the get/list response(s). be sure to use the asset id not the user id.)
